### PR TITLE
Update KSP to 1.4.30-1.0.0-alpha02 and Kotlin to 1.4.30

### PR DESCRIPTION
**Pull request by ZacSweers**
This includes some fixes to tests that would otherwise end up in an infinite loop with its new multiple rounds behavior.

### CODE REVIEW

File: ksp/src/main/kotlin/com/tschuchort/compiletesting/Ksp.kt [hunk 1]
```diff
@@ -102,14 +102,16 @@ private val KotlinCompilation.kspCachesDir: File
...
 ) : AbstractKotlinSymbolProcessingExtension(
     options = options,
     logger = logger,
     testMode = false
 ) {
-    override fun loadProcessors() = processors
+    private val loadedProcessors = processors
+
+    override fun loadProcessors() = loadedProcessors
 }
```
Comment (tschuchortdev) [inline ksp/src/main/kotlin/com/tschuchort/compiletesting/Ksp.kt:115]: Does this make any difference?

**Review by tschuchortdev — APPROVED** [review]

**Comment by tschuchortdev** [comment]
Thanks for this contribution! I think you will have to rebase it yourself because I don't have write access to your fork.

File: ksp/src/main/kotlin/com/tschuchort/compiletesting/Ksp.kt (hunk 1 above)
Comment (ZacSweers) [reply ksp/src/main/kotlin/com/tschuchort/compiletesting/Ksp.kt:115]: Yes there was a signature change that the previous version conflicted with

**Comment by ZacSweers** [comment]
@tschuchortdev done, sorry for the delay. You should have access to push to PR branches though for future ref

image

**Pull request merged**